src/site/content/en/blog/url-protocol-handler/index.md

```diff
@@ -48,49 +48,220 @@ The secure-content-only `Navigator` method
+| 3. Gather feedback & iterate on design   | [In progress](#feedback) |
+| 4. Origin trial                          | Not started              |
+| 5. Launch                                | Not started              |
+
+</div>
+
+### Enabling via chrome://flags
+
+To experiment with URL protocol handler registration for PWAs locally, without an origin trial
+token, enable the `#enable-desktop-pwas-protocol-handling` flag in `chrome://flags`.
+
+## How to use URL protocol handler registration for PWAs
+
+The API of the URL protocol handler registration for PWAs feature is modeled closely along the
+existing API of `navigator.registerProtocolHandler()`. Just this time the information is passed
+declaratively via the Web Application Manifest in a new property `"protocol_handlers"` that takes an
+array of objects with the two required keys `"protocol"` and `"url"`. The code snippet below shows
+the relevant excerpt of a Web Application Manifest that would register the protocols `web+tea` and
+`web+coffee`. The values are strings containing the URL of the handler with the required `%s`
+placeholder that will be replaced with the escaped URL to be handled.
+
+```json
+{
+  "protocol_handlers": [
+    {
+      "protocol": "web+tea",
+      "url": "/tea?type=%s"
+    },
+    {
+      "protocol": "web+coffee",
+      "url": "/coffee?type=%s"
+    }
+  ]
+}
+```
+
+### Multiple apps registering for the same protocol
+
+If multiple applications register themselves as handlers for the same scheme, for example, the
+`mailto` protocol, the operating system will show the user a picker and let them decide which of the
+registered handlers to use.
+
+### The same app registering for multiple protocols
+
+The same app can register itself for multiple protocols, as you can see in the code sample above.
+
+### App updates and handler registration
+
+Handler registrations are synchronized with the latest manifest version provided by the app. There
+are two cases:
+
+1. An update that adds new handlers triggers handler registration (separate from app installation).
+1. An update that removes handlers triggers handler unregistration (separate from app
+   uninstallation).
+
+## Demo
+
+You can see a demo of URL protocol handler registration for PWAs on Glitch.
+
+1. Go to [https://protocol-handler.glitch.me/](https://protocol-handler.glitch.me/), install the
+   PWA, and reload the app after the installation. The browser has now registered the PWA as a
+   handler for the `web+coffee` protocol with the operating system.
+1. In the installed PWA window, click on the link
+   [https://protocol-handler-link.glitch.me/](https://protocol-handler-link.glitch.me/). This will
+   open a new browser tab with three links. Click on the first or the second (latte macchiato or
+   americano). The browser will now show you a prompt and ask if you are fine with the app being a
+   protocol handler for the `web+coffee` protocol. If you agree, the PWA will open and show the
```

Comment:
We're really asking users to judge whether schemes can be used?

Reply:
Or is it presented in a non-technical way?

Reply:
This is what it looks like:

<img width="313" alt="Screen Shot 2021-05-06 at 10 00 47" src="https://user-images.githubusercontent.com/145676/117262839-f6bb8300-ae51-11eb-8164-9fb2a007de8e.png">

Note that the pitch for the feature the article is about is that there would be _no_ prompt, but the installation per se would be seen as the user being fine with the registration happening.

Reply:
(To clarify: the image above is for the JS-based registration.)

Reply:
That's correct, this is for JS-based registration but we will have a similar experience on the first time a protocol handler for a web app is invoked. This was a requirement from security/privacy folks.

![image](https://user-images.githubusercontent.com/1686833/117515279-a116d180-af4a-11eb-9272-40b9e41dd23c.png)

Reply:
[Edit 1]: I now see this is also below. Apologies. I'm reviewing this as I read it :-)